File: Semestrul 4/MPP2/Problema8_SC_REST/Problema8SC_Rest/Problema8Services/src/main/java/problema8/services/IObserver.java
package problema8.services;

import problema8.model.Child;
import problema8.model.Enrollment;
import problema8.model.User;

import java.util.List;

public interface IObserver {
    void userLoggedIn(User user) throws ProbExceptions;
    void userLoggedOut(User user) throws ProbExceptions;
    void getChildrenByEvent(List<Child> childrens) throws ProbExceptions;
    void saveChildAndEnrollment(Child child, Enrollment enrollment) throws ProbExceptions;
}

File: Semestrul 4/MPP2/Problema8_Csharp_ORM/Services/IObserver.cs
using Problema8SC_CSharp.Model;

namespace Problema8SC_CSharp.Services;

public interface IObserver
{
    void UserLoggedIn(User user);
    void UserLoggedOut(User user);
    void GetChildrenByEvent(List<Child> children);
    void SaveChildAndEnrollment(Child child, Enrollment enrollment);

}